executor/include/executor/trait/is_executor.hpp

```diff
@@ -15,34 +15,48 @@
 
 namespace executor {
 
-template <class Executor, typename = void>
-struct is_executor : std::false_type {};
-
 namespace detail {
 const auto noop = [] {};
 
-template <typename Executor, typename = void>
+/**
+ * \brief Checks if the given type T provides a member function named `execute` that takes a
+ * nullary callable
+ */
+template <typename T, typename = void>
 struct contains_execute : std::false_type {};
 
-template <typename Executor>
+template <typename T>
 struct contains_execute<
-    Executor,
-    std::enable_if_t<std::is_same<
-        decltype(std::declval<Executor>().execute(detail::noop)), void>::value>>
+    T, std::enable_if_t<std::is_same<
+           decltype(std::declval<T>().execute(detail::noop)), void>::value>>
     : std::true_type {};
 
 }  // namespace detail
 
-// Part of Proposal P0443R10: Removed from R11 in favour on concepts
-template <typename Executor>
-struct is_executor<
-    Executor,
-    std::enable_if_t<std::is_copy_constructible<Executor>::value &&
-                     detail::contains_execute<Executor>::value &&
-                     executor::equality_comparable<Executor, Executor>::value>>
+/**
+ * \brief A given type T is an Executor if it satisfies the following
+ * properties:
+ *  1. Provides a function named execute that eagerly submits work on a single
+ * execution agent created for it by the executor
+ *  2. Is a CopyConstructible type
+ *  3. Is a EqualityComparable type
+ *
+ *  This concept was finalized in P1660R0 and merged in the draft P0443R13 in
+ * the form of a concept
+ *
+ *  \todo Convert to a concept in C++20
+ */
+template <class T, typename = void>
+struct is_executor : std::false_type {};
+
+template <typename T>
+struct is_executor<T,
+                   std::enable_if_t<std::is_copy_constructible<T>::value &&
+                                    detail::contains_execute<T>::value &&
+                                    executor::equality_comparable<T, T>::value>>
     : std::true_type {};
 
-template <typename Executor>
-constexpr bool is_executor_v = is_executor<Executor>::value;
+template <typename T>
+constexpr bool is_executor_v = is_executor<T>::value;
```

Comment:
static?

Reply:
Made it static const. constexpr not possible with lambda pre C++17 due to it being a non-literal type.